File: ds/array/dynamic-array.js
function dynamicArray(n, queries) {
  // Write your code here
  let lastAnswer = 0;
  const result = [];
  const arr = [];
  for (let i = 0; i < n; i++) {
    arr[i] = [];
  }

  queries.forEach((queryArr) => {
    const idx = (queryArr[1] ^ lastAnswer) % n;
    if (queryArr[0] === 1) {
      arr[idx].push(queryArr[2]);
    } else {
      lastAnswer = arr[idx][queryArr[2] % arr[idx].length];
      result.push(lastAnswer);
    }
  });
  return result;
}
dynamicArray(2, [
  [1, 0, 5],
  [1, 1, 7],
  [1, 0, 3],
  [2, 1, 0],
  [2, 1, 1],
]);

File: ds/array/dynamic-array.py
def dynamic_array(n, queries):
  result = []
  arr = [[]]*n
  answer = 0

  print(arr)

  for query in queries:
    idx = (query[1] ^ answer) % n
    if query[0] == 1:
      arr[idx].append(query[2])
    else:
      answer = arr[idx][query[2] % len(arr[idx])]
      result.append(answer)
  return result

result = dynamic_array(2, [
  [1, 0, 5],
  [1, 1, 7],
  [1, 0, 3],
  [2, 1, 0],
  [2, 1, 1],
])

print(result)